lib/plugins/acts_as_activity_provider/lib/acts_as_activity_provider.rb

```diff
@@ -191,13 +191,13 @@ def restrict_projects_by_user(options, user, query)
               allowed_projects = []
 
               user.projects_by_role.each do |role, projects|
-                allowed_projects << projects.map(&:id) if role.allowed_to?(perm.name)
+                allowed_projects << projects.map(&:id) if perm && role.allowed_to?(perm.name)
```

Comment:
it allows me to see the project when i have a role that has a certain permission. if that permission is undefined/missing how would it make sense to allow me to see that project? i'd believe there's some other permission that eventually gives me the right to see the project. if not, i'm out. but yeah, who can know for sure in that code jungle. i think @ulferts can add some valuable thoughts here and finally i would just rely on @travisci.